### PR TITLE
values: removal of .global field

### LINKED ISSUE #5425: viz: Add Control plane tracing
## Feature Request

#5340 removed those templates from viz extension, and they should be added back.

### LINKED ISSUE #5620: Removal of `.global` in values
## Feature Request

Currently, As chart dependencies are not used anymore, The `.global` field can be removed to make these values further simple. Should we do this for `2.10` as there are already some changes in values and this could be in line with the simplification?

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #5620 

With the new extension model, We no longer need `Global` field
as we don't rely on chart dependencies anymore. This helps us
further cleanup Values, and make the configuration simpler.

To make upgrades and the usage of new CLI with older config work,
We add a new method called `config.RemoveGlobalFieldIfPresent` that
is used in the `upgrade` and `FetchCurrentConfiguration` paths to remove
`global` field and attach its child nodes if `global` is present. This is verified
by the `TestFetchCurrentConfiguration`'s older test that has the `global`
field.

We also don't yet remove `.global` in some helm stable-upgrade tests for
the initial install to work.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

**Review by alpeb — COMMENTED** [review]
Awesome :sunglasses: 

In the PR description I think you meant linking to #5620 :wink: 

There are still references to `global` in `cli/cmd/testdata/install_control-plane.golden` but it looks like that file isn't being used by any test anymore. Do you mind deleting that file as part of this PR?

Looking forward to the integration tests updates :-)

**Review by alpeb — COMMENTED** [review]
IIUC, `linkerd upgrade` will automatically migrate `global` values into the root? That's cool.

File: pkg/healthcheck/healthcheck_test.go [hunk 1]
```diff
@@ -2242,6 +2242,188 @@ func TestFetchCurrentConfiguration(t *testing.T) {
...
+  name: linkerd-config
+  namespace: linkerd
+data:
+  global: |
+    {"linkerdNamespace":"linkerd","cniEnabled":false,"version":"install-control-plane-version","identityContext":{"trustDomain":"cluster.local","trustAnchorsPem":"fake-trust-anchors-pem","issuanceLifetime":"86400s","clockSkewAllowance":"20s"}}
+  proxy: |
+    {"proxyImage":{"imageName":"ghcr.io/linkerd/proxy","pullPolicy":"IfNotPresent"},"proxyInitImage":{"imageName":"ghcr.io/linkerd/proxy-init","pullPolicy":"IfNotPresent"},"controlPort":{"port":4190},"ignoreInboundPorts":[],"ignoreOutboundPorts":[],"inboundPort":{"port":4143},"adminPort":{"port":4191},"outboundPort":{"port":4140},"resource":{"requestCpu":"","requestMemory":"","limitCpu":"","limitMemory":""},"proxyUid":"2102","logLevel":{"level":"warn,linkerd=info"},"disableExternalProfiles":true,"proxyVersion":"install-proxy-version","proxy_init_image_version":"v1.3.8","debugImage":{"imageName":"ghcr.io/linkerd/debug","pullPolicy":"IfNotPresent"},"debugImageVersion":"install-debug-version"}
+  install: |
+    {"cliVersion":"dev-undefined","flags":[]}
+  values: |
```
[the remark this answers is not in the document]
Comment (Pothulapati) [reply pkg/healthcheck/healthcheck_test.go:2255]: Ahhhh, Okay, There is the same test above which has the same test case but along with `global` field which works after the new conversion logic. This was present previously.

This test checks if this works without `global`

[the remark this answers is not in the document]
Comment (alpeb) [reply pkg/healthcheck/healthcheck_test.go:2255]: ok :cool:

**Comment by alpeb** [comment]
It doesn't make sense to add a `RemoveGlobalFieldIfPrePresent()` call into `loadStoredValuesLegacy()` because legacy `linkerd-config` doesn't have `values` per se, right?

**Review by adleong — APPROVED** [review]
This change is a scary, but worked well in my testing.  One thing to note is that users who upgrade and then try to use an old version of the CLI will see strange behavior because the old version of the CLI will be looking for the globals field.  I think this is okay, but just something to be aware of.

**Comment by Pothulapati** [comment]
@alpeb Yep, You are right. That format is different and [with the changes in `config.ToValues`](https://github.com/linkerd/linkerd2/pull/5699/files#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5R159) we are actually converting it from the older format I.e config into the current format i.e values without globals directly.

**Review by alpeb — APPROVED** [review]
Thanks @Pothulapati !

**Pull request merged**